integration_test/gce/gce_testing.go

```diff
@@ -739,6 +739,95 @@ func runCommand(ctx context.Context, logger *log.Logger, stdin io.Reader, args [
 	return output, err
 }
 
+// getGcloudConfigDir returns the current gcloud configuration directory.
+func getGcloudConfigDir(ctx context.Context) (string, error) {
+	null, err := os.Open(os.DevNull)
+	if err != nil {
+		panic(fmt.Sprintf("unable to open os.DevNull: %v", err))
+	}
+	out, err := RunGcloud(ctx, log.New(null, "", 0), "", []string{"info", "--format=value[terminator=''](config.paths.global_config_dir)"})
```

Comment:
If we run this once in init() and save the result, we don't have to run it a bunch of times and don't have to have extra code for logging the output to /dev/null either.

Plus, we can do whatever we need to do to "warm up" the configuration directory on Kokoro, so that gcloud doesn't need to regenerate the same 4 files hundreds of times per test run. Unless the "relevantFiles" are already generated somehow by this point when running on Kokoro...

Reply:
The whole point is to create a unique per-test configuration directory, to prevent data races between tests. I agree that _within a test_ we shouldn't have to re-create the directory. But this is only invoked in code that is used at the start of the tests…

Reply:
fair enough